src/NuGet.Clients/NuGet.CommandLine/Commands/SourcesCommand.cs

```diff
@@ -104,34 +111,57 @@ private void EnableOrDisableSource(bool enabled)
 
         private void RemoveSource()
         {
-            if (String.IsNullOrEmpty(Name))
+            if (string.IsNullOrEmpty(Name))
             {
                 throw new CommandLineException(LocalizedResourceManager.GetString("SourcesCommandNameRequired"));
             }
             // Check to see if we already have a registered source with the same name or source
             var sourceList = SourceProvider.LoadPackageSources().ToList();
-            var matchingSources = sourceList.Where(ps => String.Equals(Name, ps.Name, StringComparison.OrdinalIgnoreCase)).ToList();
+            var matchingSources = sourceList.Where(ps => string.Equals(Name, ps.Name, StringComparison.OrdinalIgnoreCase)).ToList();
             if (!matchingSources.Any())
             {
                 throw new CommandLineException(LocalizedResourceManager.GetString("SourcesCommandNoMatchingSourcesFound"), Name);
             }
 
+            if (Trust)
+            {
+                matchingSources.ForEach(p => SourceProvider.DeleteTrustedSource(p.Name));
+            }
+            else
+            {
+                var trustedSourceList = matchingSources.Where(p => p.TrustedSource != null).Select(p => UpdateServiceIndexTrustedSource(p));
+                SourceProvider.SaveTrustedSources(trustedSourceList);
+            }
+
             sourceList.RemoveAll(matchingSources.Contains);
             SourceProvider.SavePackageSources(sourceList);
             Console.WriteLine(LocalizedResourceManager.GetString("SourcesCommandSourceRemovedSuccessfully"), Name);
         }
 
-        private void AddNewSource()
+        private TrustedSource UpdateServiceIndexTrustedSource(PackageSource source)
+        {
+            if (source.TrustedSource != null)
+            {
+                var trustedSource = source.TrustedSource;
+                trustedSource.ServiceIndex = new ServiceIndexTrustEntry(source.Source);
+
+                return trustedSource;
+            }
+
+            return null;
```

Comment:
Do you have a test for when `null` is returned?

Reply:
added test for this case